src/main/java/com/github/lordrex34/config/model/ConfigFieldInfo.java

```diff
@@ -0,0 +1,206 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.github.lordrex34.config.model;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.lordrex34.config.ConfigManager;
+import com.github.lordrex34.config.annotation.ConfigField;
+import com.github.lordrex34.config.annotation.ConfigGroupBeginning;
+import com.github.lordrex34.config.annotation.ConfigGroupEnding;
+import com.github.lordrex34.config.postloadhooks.IConfigPostLoadHook;
+import com.github.lordrex34.config.supplier.IConfigValueSupplier;
+import com.github.lordrex34.config.util.PropertiesParser;
+
+/**
+ * @author NB4L1 (original concept)
+ * @author lord_rex
+ */
+public final class ConfigFieldInfo
+{
+	private static final Logger LOGGER = LoggerFactory.getLogger(ConfigFieldInfo.class);
+	
+	/** To avoid creating the same value supplier thousand times. */
+	private static final Map<String, IConfigValueSupplier<?>> VALUE_SUPPLIERS = new HashMap<>();
+	
+	/** To avoid creating the same post load hook thousand times. */
+	private static final Map<String, IConfigPostLoadHook> POST_LOAD_HOOKS = new HashMap<>();
+	
+	/** The class that is being scanned. */
+	private final Class<?> _clazz;
+	
+	/** The field that contains {@link ConfigField} annotation. */
+	private final Field _field;
+	
+	/** The configuration field information annotation. */
+	private final ConfigField _configField;
+	
+	/** Group beginning marker annotation. */
+	private final ConfigGroupBeginning _beginningGroup;
+	
+	/** Group ending marker annotation. */
+	private final ConfigGroupEnding _endingGroup;
+	
+	/**
+	 * Constructs a new information container class for the field.
+	 * @param clazz the class that is being scanned
+	 * @param field the field that contains {@link ConfigField} annotation
+	 */
+	public ConfigFieldInfo(Class<?> clazz, Field field)
+	{
+		_clazz = clazz;
+		_field = field;
+		_configField = _field.getDeclaredAnnotation(ConfigField.class);
+		_beginningGroup = _field.getDeclaredAnnotation(ConfigGroupBeginning.class);
+		_endingGroup = _field.getDeclaredAnnotation(ConfigGroupEnding.class);
+	}
+	
+	/**
+	 * Loads and configures the field with its proper values.
+	 * @param configPath path of the properties file
+	 * @param properties the regular properties
+	 * @param overriddenProperties the content of {@code override.properties}
+	 */
+	public void load(Path configPath, PropertiesParser properties, PropertiesParser overriddenProperties)
+	{
+		// Skip inappropriate fields.
+		if (!Modifier.isStatic(_field.getModifiers()) || Modifier.isFinal(_field.getModifiers()))
+		{
+			LOGGER.debug("Skipping non static or final field: {}#{}", _clazz.getSimpleName(), _field.getName());
```

Comment:
(Erm.. dunno. The "original use case" can have many abnormalities I fear to comprehend... if you know what I mean.)

Reply:
(the original use-case contained only static fields nothing else... it could have static final for constants... both acceptable... every non-static seems concept-breaking...)

Reply:
Done.